tests/Makefile

```diff
@@ -0,0 +1,250 @@
+
+CARTESI_IMAGES_PATH ?= $(abspath ./build/images)
+CARTESI_TESTS_PATH ?= $(abspath ./build/machine)
+CARTESI_TESTS_UARCH_PATH ?= $(abspath ./build/uarch)
+
+export CARTESI_IMAGES_PATH
+export CARTESI_TESTS_PATH
+export CARTESI_TESTS_UARCH_PATH
+
+TARGET_OS ?= $(shell uname -s)
+ifeq ($(TARGET_OS),Darwin)
+LIBCARTESI_SO=libcartesi-$(MACHINE_EMULATOR_SO_VERSION).dylib
+LIBCARTESI_SO_GRPC=libcartesi_grpc-$(MACHINE_EMULATOR_SO_VERSION).dylib
+LIBCARTESI_SO_JSONRPC=libcartesi_jsonrpc-$(MACHINE_EMULATOR_SO_VERSION).dylib
+
+NUM_JOBS := $(shell sysctl -n hw.ncpu)
+
+LIBASAN_SO=$(shell realpath `$(CC) -print-file-name=libclang_rt.asan_osx_dynamic.dylib`)
+LD_PRELOAD_PREFIX=DYLD_INSERT_LIBRARIES="$(LIBASAN_SO)"
+else
+LIBCARTESI_SO=libcartesi-$(MACHINE_EMULATOR_SO_VERSION).so
+LIBCARTESI_SO_GRPC=libcartesi_grpc-$(MACHINE_EMULATOR_SO_VERSION).so
+LIBCARTESI_SO_JSONRPC=libcartesi_jsonrpc-$(MACHINE_EMULATOR_SO_VERSION).so
+
+NUM_JOBS := $(shell nproc)
+
+LIBASAN_SO=$(shell realpath `$(CC) -print-file-name=libasan.so`)
+LIBSTDCPP_SO=$(shell realpath `$(CC) -print-file-name=libstdc++.so`)
+LD_PRELOAD="$(LIBASAN_SO) $(LIBSTDCPP_SO)"
+LD_PRELOAD_PREFIX=LD_PRELOAD=$(LD_PRELOAD)
+endif
+
+ifeq ($(coverage),yes)
+NUM_JOBS = 1
+endif
+
+COVERAGE_TOOLCHAIN?=gcc
+COVERAGE_OUTPUT_DIR=$(BUILDDIR)/coverage
+COVERAGE_WORKLOAD=\
+	dhrystone 1000000; \
+	whetstone 10000
+COVERAGE_SOURCES=$(filter-out ../src/%.pb.h, $(wildcard ../src/*.h) $(wildcard ../src/*.cpp))
+
+ifeq ($(sanitize),yes)
+LUA=$(LD_PRELOAD_PREFIX) $(LUA_BIN)
+else
+LUA=$(LUA_BIN)
+endif
+
+RUN_ARCH_TEST = $(LUA) $(abspath ./lua/run-rv64i-arch-test.lua)
+
+CARTESI_IMAGES = $(CARTESI_IMAGES_PATH)/linux.bin
+
+build-tests-all: misc build-tests-machine-with-toolchain build-tests-uarch-with-toolchain images
+
+build-tests-machine: machine riscv-tests
+
+build-tests-uarch: uarch riscv-arch-test
+
+build-tests-misc: misc
+
+build-tests-images: images
+
+build-tests-uarch-with-toolchain uarch-with-toolchain riscv-arch-test-with-toolchain: TOOLCHAIN_TAG:=0.16.0-rv64ima-lp64
+
+machine uarch misc:
+	@$(MAKE) $(BUILDDIR)/$@
+	@$(MAKE) -C $@ BUILDDIR=$(BUILDDIR)/$@
+
+riscv-tests:
+	@cd $@ && ./configure
+	@$(MAKE) -C $@ RISCV_PREFIX=$(RISCV_PREFIX)
+	@$(MAKE) copy-riscv-tests
+
+riscv-arch-test:
+	@$(MAKE) $(BUILDDIR)/uarch-$@
+	@$(MAKE) -C $@/riscv-target/cartesi TOOLCHAIN_PREFIX=$(RISCV_PREFIX) MACHINE_EMULATOR_SRC_DIR=$(MACHINE_EMULATOR_SRC_DIR)
+	@$(MAKE) -C $@ build XLEN=64 RISCV_TARGET=cartesi RISCV_ISA=rv64i RISCV_DEVICE=I TOOLCHAIN_PREFIX=$(RISCV_PREFIX) MACHINE_EMULATOR_SRC_DIR=$(MACHINE_EMULATOR_SRC_DIR) WORK=$(BUILDDIR)/uarch-$@ JOBS=-j$$(nproc)
+
+$(BUILDDIR)/%:
+	@mkdir -p $(BUILDDIR)/$*
+
+$(CARTESI_IMAGES): | $(CARTESI_IMAGES_PATH)
+	@wget -nc -i dependencies -P $(CARTESI_IMAGES_PATH)
+	@shasum -ca 256 dependencies.sha256
+	@cd $(CARTESI_IMAGES_PATH) && ln -s rootfs-tools-v0.14.1.ext2 rootfs.ext2
```

Comment:
I wish this value could be read the version itself from `dependencies` file, or we had one one place with the hard-coded version, so we minimize number of hard-coded version when making releases.

Reply:
Can you open a issue for this?